libs/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts

```diff
@@ -89,7 +89,7 @@ export class DataLayerColorService {
       for (let annotation of layer.annotations) {
         const anno = annotation as BoxAnnotationOptions;
         const label = anno.label?.content;
-        if (typeof label === 'string' && label.startsWith(dataset.label)) {
+        if (typeof label === 'string' && dataset.chartsOnFhir?.group && label.startsWith(dataset.chartsOnFhir?.group)) {
           anno.backgroundColor = this.addTransparency(this.getColor(dataset));
         }
```

Comment:
It's not intuitive that the dataset needs to define a group that matches the annotation label in order for the annotation to be colored correctly. Maybe we should add a `chartsOnFhir.referenceRangeAnnotationId` property that we can use to match the reference range?
(We could create a separate PR for this if you prefer)